gateway/api/views/jobs.py

```diff
@@ -145,10 +145,10 @@ def list(self, request):
 
             page = self.paginate_queryset(queryset)
```

Comment:
@paaragon let's use this opportunity to migrate the `queryset` too. The idea as we commented is to put all the logic related with the ORM in the Repositories. In this case the philosophy would be similar to what we did in the `list` for functions: https://github.ibm.com/IBM-Q-Software/qiskit-serverless/blob/main/gateway/api/views/programs.py#L104